pandas/core/frame.py

```diff
@@ -6243,11 +6244,10 @@ def diff(self, periods=1, axis=0):
     # Function application
 
     def _gotitem(self,
-                 key,           # type: Union[str, List[str]]
-                 ndim,          # type: int
-                 subset=None    # type: Union[Series, DataFrame, None]
-                 ):
-        # type: (...) -> Union[Series, DataFrame]
+                 key: Union[str, List[str]],
+                 ndim: int,
+                 subset: Optional[Series, 'DataFrame'] = None,
```

Comment:
I think I misstated before but should be `Optional[Union[Series, 'DataFrame']]` right?

Reply:
Yep. Apparently feeding Optional more than one item doesn't work. I've changed it as requested.

Thanks for walking me through this.